### PR TITLE
[KOGITO-3495] Error handling on RestworkItemhandler

### PR DESCRIPTION
**Pull request by fjtirado**
If rest invocation failed, workitem will be aborted.

### CODE REVIEW

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
jenkins retest this

File: kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java [hunk 1]
```diff
@@ -87,28 +88,24 @@ public void executeWorkItem(WorkItem workItem, WorkItemManager manager) {
...
         // execute request
-         Handler<AsyncResult<HttpResponse<Buffer>>> handler = event -> 
-                manager
-                .completeWorkItem(
-                                  workItem.getId(),
-                                  Collections
-                                      .singletonMap(RESULT, resultHandler.apply(inputModel, event.result().bodyAsJsonObject())));
+        Handler<AsyncResult<HttpResponse<Buffer>>> handler = event -> {
+            if (event.cause() != null) {
+                throw new WorkItemExecutionError("500", event.cause());
```
Comment (MarianMacik) [inline kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java:99]: Probably a better message would be welcome.

File: kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java [hunk 2]
```diff
@@ -87,28 +88,24 @@ public void executeWorkItem(WorkItem workItem, WorkItemManager manager) {
...
         }
         // execute request
-         Handler<AsyncResult<HttpResponse<Buffer>>> handler = event -> 
-                manager
-                .completeWorkItem(
-                                  workItem.getId(),
-                                  Collections
-                                      .singletonMap(RESULT, resultHandler.apply(inputModel, event.result().bodyAsJsonObject())));
+        Handler<AsyncResult<HttpResponse<Buffer>>> handler = event -> {
+            if (event.cause() != null) {
```
Comment (MarianMacik) [inline kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java:98]: Probably more straightforward would be

```
if (event.failed()) {
    throw new WorkItemExecutionError("500", event.cause());
}
```

File: kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java (hunk 1 above)
Comment (MarianMacik) [reply kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java:99]: As discussed, the code is expected.

File: kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java (hunk 2 above)
Comment (fjtirado) [reply kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java:98]: thats better indeed

File: kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java (hunk 1 above)
Comment (fjtirado) [reply kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java:99]: This is required by WorkItemExecution. Ill use the error returned by the rest service

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
Including @evacchi since he were working with futures and this might interest him
Since RestWorkItemHandler is implemented asynchronously, throwing an exception when the rest invocation fails will only makes the vertx thread where the hander is invoked "angry", so I think the best we can do is to abort the work item asyncronously, in the same way we are completing it asyncrhonously when the call succeed. 
In any case, the process will continue and, in the example case (two consecutive rest invocations, where the second call depends on data retrieved by first one), since the first call failed to provide required data, the second call will fail. 
Please note the failure is not reported in the post call that initiate the process. User will have to query the process ide for resolving the status

**Review by cristianonicolai — APPROVED** [review]

**Review by MarianMacik — APPROVED** [review]

**Pull request merged**